elfpy/markets/hyperdrive/hyperdrive_market.py

```diff
@@ -536,57 +542,41 @@ def latest_checkpoint_time(self) -> float:
         return latest_checkpoint / 365
 
     def apply_checkpoint(self, checkpoint_time: float, share_price: float) -> float:
-        """creates a new checkpoint if necessary."""
-        # return early if the checkpoint has already been updated.
+        """Creates a new checkpoint if necessary and closes matured positions.
+
+        Parameters
+        ----------
+        checkpoint_time: float
+            The block time for the checkpoint to be created or cleared.
+        share_price: float
+            The share price of the market at the checkpoint time.
+
+        Returns
+        -------
+        float
+            The share price for the checkpoint after mature positions have been closed.
+        """
+        # Return early if the checkpoint has already been updated.
         if self.market_state.checkpoints[checkpoint_time].share_price != 0 or checkpoint_time > self.block_time.time:
```

Comment:
I'm weary of this first check of `share_price` now.  We gotta talk to solidity folks I think.  If share_price is not zero, that indicates that the checkpoint in the past has already been created, so we don' try to close the checkpointed values?  seems wrong.

Reply:
Yeah I don't like that check either.

Reply:
It would be great if we clarified what it is for with the solidity team and either added a clarifying comment here or changed the conditional to be more close to what we're actually checking.

Reply:
let me ask